### PR TITLE
Address misc time picker design issues

### LINKED ISSUE #62775: TimePicker is not correct in RTL languages
timerpicker_bug

To reproduce, Set app locale to a right-to-left language (`fa` for example) then open timepicker.

Besides that "Select Time" which not translated, Why is that the time of day format is in `mm:hh` format instead of `hh:mm`? 

RTL doesn't mean everything should be written right to left, Those languages still use `hh:mm` for time format. 
Also it wasn't an issue in previous TimePicker widget.

**Flutter Doctor**
[√] Flutter (Channel beta, 1.20.0-7.3.pre, on Microsoft Windows [Version 10.0.19041.388], locale en-US)

[!] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
    X Android license status unknown.
      Try re-installing or updating your Android SDK Manager.
      See https://developer.android.com/studio/#downloads or visit
      https://flutter.dev/docs/get-started/install/windows#android-setup for detailed instructions.
[√] Chrome - develop for the web
[√] Android Studio (version 4.0)
[√] VS Code, 64-bit edition (version 1.47.3)
[√] Connected device (4 available)

### PR DESCRIPTION
**Pull request by rami-a**
## Description

This change makes small tweaks to time picker to ensure it matches the design spec (to be published soon!)

- Using bodyText1 on the dial numbers instead of subtitle1
- Making sure dial numbers use color scheme values
- Limit textfield entry to 2 characters
- Ensure providing a custom inputDecorationTheme still allows the focused fill color behavior
- Ensure that when locale is RTL, the hours/minutes do not get swapped

## Related Issues

Fixes #62775

## Tests

I added the following tests:

- Updated existing tests

## Checklist

Before you create this PR, confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*
   - [ ] I wrote a design doc: https://flutter.dev/go/template *Replace this with a link to your design doc's short link*
   - [ ] I got input from the developer relations team, specifically from: *Replace with the names of who gave advice*
   - [ ] I wrote a migration guide: https://flutter.dev/go/breaking-changes-template *Replace this with a link to a pull request that adds the migration guide to https://flutter.dev/docs/release/breaking-changes*

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: packages/flutter/lib/src/material/time_picker.dart [hunk 1]
```diff
@@ -1576,10 +1576,13 @@ class _HourMinuteTextFieldState extends State<_HourMinuteTextField> {
         child: TextFormField(
           expands: true,
           maxLines: null,
+          inputFormatters: <TextInputFormatter>[
```
Comment (willlarche) [inline packages/flutter/lib/src/material/time_picker.dart:1579]: 1 line?

File: packages/flutter/lib/src/material/time_picker.dart [hunk 2]
```diff
@@ -1821,7 +1824,7 @@ class _TimePickerDialogState extends State<_TimePickerDialog> {
             theme.colorScheme.brightness == Brightness.dark ? 1.0 : 0.6,
           ),
           onPressed: _handleEntryModeToggle,
-          icon: Icon(_entryMode == TimePickerEntryMode.dial ? Icons.keyboard : Icons.access_time),
+          icon: Icon(_entryMode == TimePickerEntryMode.dial ? Icons.keyboard_outlined : Icons.access_time),
```
Comment (willlarche) [inline packages/flutter/lib/src/material/time_picker.dart:1827]: Outlined for M2?

**Review by willlarche — COMMENTED** [review]
LGTM

File: packages/flutter/lib/src/material/time_picker.dart (hunk 1 above)
Comment (rami-a) [reply packages/flutter/lib/src/material/time_picker.dart:1579]: maxLines has to be null for the `expands` logic to work, so we limit to 2 characters

File: packages/flutter/lib/src/material/time_picker.dart (hunk 2 above)
Comment (rami-a) [reply packages/flutter/lib/src/material/time_picker.dart:1827]: yep that is expected

**Review by johnsonmh — APPROVED** [review]

**Review by HansMuller — APPROVED** [review]
LGTM however these changes are likely to break golden image tests.

**Comment by rami-a** [comment]
I've disabled my internal golden tests while I work through these changes internally and externally. So I think it will be ok to land this

**Review by HansMuller — APPROVED** [review]

**Comment by pcsosinski** [comment]
removing CP label in favor of https://github.com/flutter/flutter/issues/62775

**Pull request merged**